File: 1_easy/1001_2000/1201_1400/01317_convert_integer_to_the_sum_of_two_no_zero_integers/solution.cpp
class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        for(int i{1}; i < n; ++i) {
            int nums[] = {i, n-i};
            bool is_valid = true;
            for(auto a: nums) {
                if(! is_valid) {
                    break;
                }
                while(a > 0) {
                    if(a % 10 == 0) {
                        is_valid = false;
                        break;
                    }
                    a /= 10;
                }
            }
            if(is_valid){
                return vector<int>{i, n-i};
            }
        }
        return vector<int>{};
    }
};

File: 1_easy/1001_2000/1201_1400/01317_convert_integer_to_the_sum_of_two_no_zero_integers/solution.rs
impl Solution {
    pub fn get_no_zero_integers(n: i32) -> Vec<i32> {
        for i in 1..n {
            let nums = [i,n-i];
            let mut is_valid = true;
            for mut a in nums {
                if ! is_valid {
                    break;
                }
                while a > 0 {
                    if a % 10 == 0 {
                        is_valid = false;
                        break;
                    }
                    a /= 10;
                }
            }
            if is_valid {
                return vec![i, n-i];
            }
        }
        Vec::new()
    }
}
